tests/app/helpers.py

```diff
@@ -49,15 +49,16 @@ def setup(self):
     def do_not_provide_access_token(self):
         self.app.wsgi_app = self.app.wsgi_app.app
 
-    def create_index(self, index_name=None):
+    def create_index(self, index_name=None, expect_success=True):
         if index_name is None:
             index_name = self.default_index_name
 
         response = self.client.put('/{}'.format(index_name), data=json.dumps({
             "type": "index",
             "mapping": self.default_mapping_name,
         }), content_type="application/json")
-        assert response.status_code in (200, 201)
+        if expect_success:
```

Comment:
I'm not sure I see the point of this parameter - we're returning the response, so if you want to expect success, you can just assert it?

Reply:
Yeah... don't we already call `assert_response_status` in the tests?

Reply:
In the particular test we're tinkering with, yes, but not every place that calls `create_index` does perform its own assertions and this will catch any failures then.

But really, I'm just trying to keep my changes minimal. The test I've modified is the only place that `create_index` isn't expected to succeed, so keeping this in place for the other n invocations makes sense.